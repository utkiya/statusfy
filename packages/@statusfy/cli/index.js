// Inspired on https://github.com/vuejs/vuepress/blob/master/packages/%40vuepress/cli/index.js
const program = require('commander')
const satisfies = require('semver/functions/satisfies')
const { chalk, path } = require('@statusfy/common')

const packageError = (name) => {
  console.log(chalk.red(
    `\n[statusfy] @statusfy/cli ` +
    `requires ${name} to be installed.\n` +
    `${err}`
  ))
}

try {
  require.resolve('@statusfy/core')
} catch (err) {
  packageError('@statusfy/core')
  process.exit(1)
}

try {
  require.resolve('@statusfy/editor')
} catch (err) {
  packageError('@statusfy/editor')
  process.exit(1)
}

const pkg = require('@statusfy/core/package.json')
const requiredVersion = pkg.engines.node

if (!satisfies(process.version, requiredVersion)) {
  console.log(chalk.red(
    `Minimum Node version not met:\n` +
    `You are using Node ${process.version}, ` +
    `but Statusfy requires Node ${requiredVersion}.`
  ))
  console.log(chalk.red(`Please upgrade your Node version.`))

  process.exit(1)
}

const {
  init,
  dev,
  build,
  generate,
  start,
  newIncident,
  deleteIncident,
  updateIncident
} = require('@statusfy/core/lib')
const sourceDir = path.resolve('.')

program
  .version(pkg.version)
  .usage('<command> [options]')

program
  .command('init')
  .description('Create a base project')
  .option('-d, --dir <outDir>', 'specify the installation directory')
  .action(({ dir }) => {
    const outDir = dir ? path.resolve(dir) : null
    wrapCommand(init)(sourceDir, { outDir })
  })

program
  .command('dev')
  .description('Starts the application in development mode (hot-code reloading, error reporting, etc.).')
  .option('-p, --port <port>', 'use specified port (default: 3000)')
  .option('-s, --ssr', 'force SSR')
  .action(({ port, ssr }) => {
    wrapCommand(dev)(sourceDir, { port, ssr })
  })

program
  .command('build')
  .description('Compiles the application for production deployment')
  .option('-a, --analyze', 'launch the final bundle analysis')
  .action(({ analyze }) => {
    wrapCommand(build)(sourceDir, { analyze })
  })

program
  .command('generate')
  .description('Generate a static web application (server-rendered)')
  .option('-d, --dest <outDir>', 'specify generate output dir (default: ./dist)')
  .option('-a, --analyze', 'launch the final bundle analysis')
  .action(({ dest, analyze }) => {
    const outDir = dest ? path.resolve(dest) : null
    wrapCommand(generate)(sourceDir, { outDir, analyze })
  })

program
  .command('start')
  .description(`Starts the application in production mode. The application should be compiled with ${chalk.cyan(`statusfy build`)} first.`)
  .option('-p, --port <port>', 'use specified port (default: 3000)')
  .option('-H, --host <host>', 'use specified host (default: 127.0.0.1)')
  .action(({ host, port }) => {
    wrapCommand(start)(sourceDir, { host, port })
  })

program
  .command('new-incident')
  .description(`Creates a new incident after answering a few questions.`)
  .action(() => {
    wrapCommand(newIncident)(sourceDir, { })
  })

program
  .command('delete-incident')
  .description(`Delete an incident including translations.`)
  .action(() => {
    wrapCommand(deleteIncident)(sourceDir, { })
  })


program
  .command('update-incident')
  .description(`Update an incident including translations.`)
  .action(() => {
    wrapCommand(updateIncident)(sourceDir, {})
  })

// output help information on unknown commands
program
  .arguments('<command>')
  .action((cmd) => {
    program.outputHelp()
    console.log(`  ` + chalk.red(`Unknown command ${chalk.yellow(cmd)}.\n`))
  })

// add some useful info on help
program.on('--help', () => {
  console.log(`\n  Run ${chalk.cyan(`statusfy <command> --help`)} for detailed usage of given command.\n`)
})

program.commands.forEach(c => c.on('--help', () => console.log()))
program.parse(process.argv)

if (!process.argv.slice(2).length) {
  program.outputHelp()
}

function wrapCommand (fn) {
  return (...args) => {
    return fn(...args).catch(err => {
      console.error(chalk.red(err.stack))
      process.exitCode = 1
    })
  }
}
