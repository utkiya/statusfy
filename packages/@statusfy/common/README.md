# @statusfy/common

> Common Utils for Statusfy
