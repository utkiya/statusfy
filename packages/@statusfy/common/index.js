const path = require('upath')
const esm = require('./lib/esm')

exports.logger = require('./lib/logger')
exports.validator = require('./lib/validator')
exports.grayMatter = require('./lib/gray-matter')
exports.slugify = require('./lib/slugify')
exports.generateDemoContent = require('./lib/generate-content')
exports.style = require('./lib/style')
exports.postcss = require('./lib/postcss')
exports.esm = esm
exports.Dates = esm(path.join(__dirname, './lib/dates.js')).default

exports.url = require('url')
exports.toml = require('toml')
exports.tomlify = require('tomlify-j0.4')
exports.yaml = require('yaml')

exports.chalk = require('chalk')
exports.fse = require('fs-extra')
exports.hash = require('hash-sum')
exports.path = path
exports.LRU = require('lru-cache')
