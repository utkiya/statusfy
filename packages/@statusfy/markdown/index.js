const emoji = require('markdown-it-emoji')

const containers = require('./containers')
const frontmatter = require('./frontmatter')
const links = require('./links')

module.exports = ({ markdown = {} } = {}) => {
  const md = require('markdown-it')({
    html: true
  })
  // custom plugins
  .use(containers)
  .use(frontmatter)
  .use(links)

  // 3rd party plugins
  .use(emoji)

  // apply user config
  if (markdown.config) {
    markdown.config(md)
  }

  return md
}
