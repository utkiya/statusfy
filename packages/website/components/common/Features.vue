<template>
  <section
    id="features"
    :style="`background-image: url(${require('~/assets/img/dividers/wave-curve.svg')});`"
    class="bg-bottom bg-repeat-x bg-black py-6 z-10"
  >
    <div class="container mt-4 mb-8">
      <div class="text-center text-white mb-8">
        <h2
          class="text-3xl sm:text-3xl font-semibold leading-none mb-4"
        >
          {{ $t('features.title') }}
        </h2>

        <p
          class="max-w-md mx-auto text-xl sm:text-2xl text-blue-darker font-normal leading-tight mb-8"
        >
          {{ $t('features.description') }}
        </p>
      </div>

      <div
        class="flex flex-wrap lg:text-center bg-white rounded my-6 shadow p-6 h-full"
      >
        <div
          v-for="(feature, key) in features"
          :key="key"
          class="w-full sm:w-1/2 lg:w-1/3 px-6 py-2"
        >
          <div
            :class="`text-${feature.color}`"
            class="mb-8"
          >
            <svgicon
              :name="`fortawesome/${feature.icon}`"
              class="align-middle svg-inline--fa fa-3x"
              heigth="48"
              width="auto"
            />
          </div>
          <h3 class="text-2xl font-semibold leading-none mb-3">
            {{ $t(`features.${key}.title`) }}
          </h3>
          <!-- eslint-disable vue/no-v-html -->
          <p
            class="text-lg leading-normal text-grey-darkest mb-8"
            v-html="$t(`features.${key}.description`)"
          />
          <!-- eslint-enable -->
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import '~/components/icons/fortawesome/database-solid'
import '~/components/icons/fortawesome/mobile-alt-solid'
import '~/components/icons/fortawesome/js-square-brands'
import '~/components/icons/fortawesome/language-solid'
import '~/components/icons/fortawesome/paint-roller-solid'
import '~/components/icons/fortawesome/globe-solid'

export default {
  data() {
    return {
      features: {
        markdown: {
          icon: 'database-solid',
          color: 'blue'
        },
        pwa: {
          icon: 'mobile-alt-solid',
          color: 'purple'
        },
        stack: {
          icon: 'js-square-brands',
          color: 'orange'
        },
        customization: {
          icon: 'paint-roller-solid',
          color: 'red'
        },
        i18n: {
          icon: 'language-solid',
          color: 'yellow'
        },
        hosting: {
          icon: 'globe-solid',
          color: 'green'
        }
      }
    }
  }
}
</script>
