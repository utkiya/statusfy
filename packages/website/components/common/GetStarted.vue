<template>
  <section class="bg-white py-8">
    <div class="container my-8">
      <div class="flex lg:items-center flex-col lg:flex-row lg:justify-between">
        <div class="mb-8 lg:mb-0 lg:pr-4">
          <h2 class="text-2xl font-normal leading-none mb-3">
            {{ $t('getStarted.title') }}
          </h2>
          <h3 class="text-xl text-grey-dark font-normal leading-tight">
            {{ $t('getStarted.description') }}
          </h3>
        </div>
        <div class="lg:pl-4">
          <div class="flex items-center">
            <a
              :href="$t('links.getStarted.url')"
              class="btn btn-blue mr-4"
              target="__blank"
              rel="noopener"
            >
              {{ $t('links.getStarted.title') }}
            </a>
            <a
              :href="$t('links.github.url')"
              class="btn btn-grey"
              target="__blank"
              rel="noopener"
            >
              {{ $t('links.github.title') }}
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</template>
