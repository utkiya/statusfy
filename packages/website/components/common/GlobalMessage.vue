<template>
  <div
    v-if="$t('globalMessage') !== ''"
    class="bg-orange text-white text-center py-2 px-4 h-16"
  >
    <div class="p-2 items-center leading-none flex inline-flex h-full leading-tight">
      <!-- eslint-disable vue/no-v-html -->
      <div
        class="font-semibold mr-2 flex-auto"
        v-html="$t('globalMessage')"
      />
      <!-- eslint-enable -->
    </div>
  </div>
</template>

<style scoped>
div >>> a {
  @apply text-grey-lightest underline;

  &:hover {
    @apply text-grey-lighter;
  }
}
</style>
