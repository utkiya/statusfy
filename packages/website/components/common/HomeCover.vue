<template>
  <section
    :style="`background-image: url(${require('~/assets/img/dividers/inclined-line.svg')});`"
    class="h-full bg-bottom bg-repeat-x pt-2 pb-8 bg-white"
  >
    <div class="container">
      <div
        class="flex flex-wrap items-center justify-center text-center lg:text-left"
      >
        <div class="w-full lg:w-1/3 pb-4 lg:pr-4">
          <h1
            class="text-3xl sm:text-4xl font-semibold leading-none mb-4"
          >
            {{ $t('slogan') }}
          </h1>
          <!-- eslint-disable vue/no-v-html -->
          <h2
            class="text-lg sm:text-xl text-blue-darker font-normal leading-normal mb-2"
            v-html="$t('description')"
          />
          <!-- eslint-enable -->
          <div
            class="flex flex-col sm:flex-row justify-center lg:justify-start pt-8"
          >
            <a
              :href="$t('links.getStarted.url')"
              class="btn btn-blue mb-2 sm:mb-0 sm:mr-2"
            >
              {{ $t('links.getStarted.title') }}
            </a>
            <button
              v-scroll-to="'#features'"
              class="btn btn-grey mt-2 sm:mt-0 sm:ml-2"
            >
              {{ $t('links.learnMore.title') }}
            </button>
          </div>
        </div>
        <div class="w-full lg:w-2/3 pb-4 lg:pl-4 z-20">
          <div v-lazy-container="{ selector: 'img' }">
            <img
              :title="$t('slogan')"
              :data-src="require(`~/assets/img/home-cover/home-cover-${$i18n.locale}.jpg`)"
              :data-srcset="`${require(`~/assets/img/home-cover/home-cover-${$i18n.locale}.jpg`)} 1x, ${require(`~/assets/img/home-cover/home-cover-${$i18n.locale}@2x.jpg`)} 2x`"
              src="~assets/img/home-cover/home-cover-loading.svg"
              alt="Statusfy"
              class="rounded shadow"
              width="1027"
              height="697"
            >
          </div>
        </div>
      </div>
    </div>
  </section>
</template>

<style scoped>
section {
  background-size: 100% 100px;
}
</style>
