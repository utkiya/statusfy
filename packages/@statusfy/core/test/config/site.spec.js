import { path } from "@statusfy/common";

import generateConfig from "@/core/lib/config/generate";
import loadConfig from "@/core/lib/config/load";

const tempPath = path.resolve(__dirname, "../.tmp");

describe("config:site", () => {
  const executeConfigTest = label => {
    const sourceDir = path.resolve(tempPath, label);
    const { siteConfig } = generateConfig(sourceDir, {});
    const siteConfigSourceDir = siteConfig.sourceDir;

    siteConfig.sourceDir = path.relative(__dirname, sourceDir);

    expect(path.isAbsolute(siteConfigSourceDir)).toBeTruthy();
    expect(siteConfig).toMatchSnapshot();
  };

  test("simple", () => {
    executeConfigTest("simple");
  });

  test("cli", () => {
    executeConfigTest("cli");
  });

  test("advanced", () => {
    executeConfigTest("advanced");
  });

  test("spanish", () => {
    executeConfigTest("spanish");
  });

  test("invalid", () => {
    const sourceDir = path.resolve(tempPath, "invalid");
    const loadedConfig = loadConfig(sourceDir);
    const siteConfigErrors = loadedConfig.errors;

    expect(siteConfigErrors).toMatchSnapshot();
  });

  test("invalid-subpath", () => {
    const sourceDir = path.resolve(tempPath, "invalid-subpath");
    const loadedConfig = loadConfig(sourceDir);
    const siteConfigErrors = loadedConfig.errors;

    expect(siteConfigErrors).toMatchSnapshot();
  });
});
