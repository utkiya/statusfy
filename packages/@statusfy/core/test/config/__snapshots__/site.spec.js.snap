// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config:site advanced 1`] = `
Object {
  "analytics": Object {
    "ga": "UA-XXXXXXXXX-Y",
  },
  "baseUrl": "https://demo.statusfy.co",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "frontMatterFormat": "yaml",
    "systems": Array [
      "cdn",
      "conversions",
      "site-delivery",
      "api",
    ],
  },
  "defaultLocale": "en",
  "description": "Description",
  "head": Object {
    "link": Array [
      Object {
        "color": "#3e4e88",
        "href": "/icons/safari-pinned-tab.svg",
        "rel": "mask-icon",
      },
    ],
    "meta": Array [
      Object {
        "charset": "utf-8",
      },
      Object {
        "content": "width=device-width, initial-scale=1",
        "name": "viewport",
      },
      Object {
        "content": "Statusfy",
        "name": "generator",
      },
      Object {
        "content": "Description",
        "hid": "description",
        "name": "description",
      },
    ],
    "title": "Long Title",
    "titleTemplate": "%s | Long Title",
  },
  "locales": Array [
    Object {
      "code": "en",
      "iso": "en-US",
      "name": "English",
    },
    Object {
      "code": "es",
      "iso": "es-ES",
      "name": "Espa√±ol",
    },
  ],
  "manifest": true,
  "name": "app_name",
  "notifications": Object {
    "feeds": true,
    "icalendar": true,
    "support": false,
    "twitter": false,
    "webpush": Object {
      "onesignal": Object {
        "appId": "onesignal-app-id",
      },
    },
  },
  "serviceWorker": true,
  "short_title": "Short Title",
  "sourceDir": "../.tmp/advanced",
  "theme": Object {
    "scheduled": Object {
      "position": "belowSystems",
    },
  },
  "title": "Long Title",
}
`;

exports[`config:site cli 1`] = `
Object {
  "baseUrl": "/",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "frontMatterFormat": "yaml",
    "systems": Array [
      "api",
      "cdn",
      "dns",
      "site-delivery",
    ],
  },
  "defaultLocale": "en",
  "description": "",
  "locales": Array [
    Object {
      "code": "en",
      "iso": "en-US",
      "name": "English",
    },
  ],
  "manifest": true,
  "name": "title",
  "notifications": Object {
    "feeds": true,
    "icalendar": true,
    "support": false,
    "twitter": false,
  },
  "serviceWorker": true,
  "short_title": "Title",
  "sourceDir": "../.tmp/cli",
  "theme": Object {
    "scheduled": Object {
      "position": "belowSystems",
    },
  },
  "title": "Title",
}
`;

exports[`config:site invalid 1`] = `
Array [
  "The default Front Matter Format ([33mcontent.frontMatterFormat[39m) is invalid. These are the valid formats: [36myaml, yml, toml, json[39m.",
  "The Base URL ([33mbaseUrl[39m) is invalid: //demo.statusfy.co.
Valid Example: [36mhttps://status.yourbaseurl.com[39m.",
  "The Default Locale ([33mdefaultLocale[39m) value must be included in the locales list. Current value [36mit[39m, defined codes: [36men, es[39m.",
]
`;

exports[`config:site invalid-subpath 1`] = `
Array [
  "Statusfy doesn't support deployments under a subpath ([36m/path[39m).",
]
`;

exports[`config:site simple 1`] = `
Object {
  "baseUrl": "https://simple.example.com",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "frontMatterFormat": "yaml",
    "systems": Array [
      "api",
      "cdn",
      "dns",
      "site-delivery",
    ],
  },
  "defaultLocale": "en",
  "description": "",
  "locales": Array [
    Object {
      "code": "en",
      "iso": "en-US",
      "name": "English",
    },
  ],
  "manifest": false,
  "name": "title",
  "notifications": false,
  "serviceWorker": false,
  "short_title": "Title",
  "sourceDir": "../.tmp/simple",
  "theme": Object {
    "scheduled": Object {
      "position": "belowSystems",
    },
  },
  "title": "Title",
}
`;

exports[`config:site spanish 1`] = `
Object {
  "baseUrl": "https://demo-es.statusfy.co",
  "build": Object {
    "isStatic": false,
  },
  "content": Object {
    "dir": "content",
    "frontMatterFormat": "yaml",
    "systems": Array [
      "notificaciones",
      "monitoreo",
      "entrega-sitio",
    ],
  },
  "defaultLocale": "es",
  "description": "Descripci√≥n",
  "locales": Array [
    Object {
      "code": "es",
      "iso": "es-ES",
      "name": "Espa√±ol",
    },
    Object {
      "code": "fr",
      "iso": "fr-FR",
      "name": "Fran√ßais",
    },
  ],
  "manifest": true,
  "name": "Nombre Aplicaci√≥n",
  "notifications": Object {
    "feeds": true,
    "icalendar": true,
    "support": false,
    "twitter": false,
  },
  "serviceWorker": true,
  "short_title": "T√≠tulo",
  "sourceDir": "../.tmp/spanish",
  "theme": Object {
    "scheduled": Object {
      "position": "belowSystems",
    },
  },
  "title": "T√≠tulo",
}
`;
